crates/iota-indexer/src/store/pg_indexer_store.rs

```diff
@@ -869,6 +872,49 @@ impl PgIndexerStore {
         })
     }
 
+    /// We enforce index-status semantics for checkpointed transactions
+    /// in `tx_global_order`.
+    ///
+    /// Namely, checkpointed transactions (i.e. with `optimistic_sequence_number
+    /// == 0`) are updated to `optimistic_sequence_number == -1` to indicate
+    /// that they have been persisted in the database.
+    fn update_status_for_checkpoint_transactions_chunk(
+        &self,
+        tx_order: Vec<CheckpointTxGlobalOrder>,
+    ) -> Result<(), IndexerError> {
+        let guard = self
+            .metrics
+            .checkpoint_db_commit_latency_tx_insertion_order_chunks
+            .start_timer();
+
+        let num_transactions = tx_order.len();
+        transactional_blocking_with_retry!(
+            &self.blocking_cp,
+            |conn| {
+                on_conflict_do_update_with_condition!(
+                    tx_global_order::table,
+                    tx_order.clone(),
+                    tx_global_order::tx_digest,
+                    tx_global_order::optimistic_sequence_number.eq(IndexStatus::Completed),
+                    tx_global_order::optimistic_sequence_number.eq(IndexStatus::Started),
+                    conn
+                );
+                Ok::<(), IndexerError>(())
+            },
+            PG_DB_COMMIT_SLEEP_DURATION
+        )
+        .tap_ok(|_| {
+            let elapsed = guard.stop_and_record();
+            info!(
+                elapsed,
+                "Persisted {} chunked txs insertion order", num_transactions
```

Comment:
The log here and below could be updated as the function is not about insertion order.

Reply:
df8d5d8443